csharp/test/TestCommon/TestHelper.cs

```diff
@@ -284,3 +284,260 @@ public static async Task<int> RunTestAsync<T>(string[] args) where T : TestHelpe
+                {
+                    sb.Append(host);
+                }
+                sb.Append(" -p ");
+                sb.Append(port);
+                return sb.ToString();
+            }
+        }
+
+        public static string GetTestProxy(
+            string identity,
+            Dictionary<string, string> properties,
+            int num = 0,
+            string? transport = null)
+        {
+            if (GetTestProtocol(properties) == Protocol.Ice2 && transport != "udp")
+            {
+                var sb = new StringBuilder("ice+");
+                sb.Append(transport ?? GetTestTransport(properties));
+                sb.Append("://");
+                string host = GetTestHost(properties);
+                if (host.Contains(':'))
+                {
+                    sb.Append('[');
+                    sb.Append(host);
+                    sb.Append(']');
+                }
+                else
+                {
+                    sb.Append(host);
+                }
+                sb.Append(':');
+                sb.Append(GetTestBasePort(properties) + num);
+                sb.Append('/');
+                sb.Append(identity);
+                return sb.ToString();
+            }
+            else // i.e. ice1
+            {
+                var sb = new StringBuilder(identity);
+                sb.Append(':');
+                sb.Append(transport ?? GetTestTransport(properties));
+                sb.Append(" -h ");
+                string host = GetTestHost(properties);
+                if (host.Contains(':'))
+                {
+                    sb.Append('"');
+                    sb.Append(host);
+                    sb.Append('"');
+                }
+                else
+                {
+                    sb.Append(host);
+                }
+                sb.Append(" -p ");
+                sb.Append(GetTestBasePort(properties) + num);
+                return sb.ToString();
+            }
+        }
+
+        public static string GetTestHost(Dictionary<string, string> properties)
+        {
+            if (!properties.TryGetValue("Test.Host", out string? host))
+            {
+                host = "127.0.0.1";
+            }
+            return host;
+        }
+
+        public static Protocol GetTestProtocol(Dictionary<string, string> properties)
+        {
+            if (!properties.TryGetValue("Test.Protocol", out string? value))
+            {
+                return Protocol.Ice2;
+            }
+            try
+            {
+                return ProtocolExtensions.Parse(value);
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidConfigurationException(
+                    $"invalid value for for Test.Protocol: `{value}'", ex);
+            }
+        }
+
+        public static ZeroC.Ice.Encoding GetTestEncoding(Dictionary<string, string> properties)
+            => ProtocolExtensions.GetEncoding(GetTestProtocol(properties));
```

Comment:
```suggestion
        public static ZeroC.Ice.Encoding GetTestEncoding(Dictionary<string, string> properties) =>
            ProtocolExtensions.GetEncoding(GetTestProtocol(properties));
```

Reply:
I think we should just add a `using ZeroC.Ice;`

Reply:
Fine, I was just pointing that the `=>` is out of place

Reply:
Fixed.